RestaurantReservation.Db/Configurations/OrderConfiguration.cs

```diff
@@ -22,7 +22,7 @@ public void Configure(EntityTypeBuilder<Order> builder)
             builder.HasOne(r => r.Reservation)
               .WithMany(o => o.Orders)
               .HasForeignKey(r => r.ReservationId)
-              .OnDelete(DeleteBehavior.ClientSetNull);
+              .OnDelete(DeleteBehavior.Cascade);
```

Comment:
what's the reason behind this change ?

Reply:
the change ensures that when a reservation is deleted all related order are also deleted automaticaly this behavior maintains data integrity by preventing orphaned records and ensuring that the system remains consistent as each order should not exist without a valid reservation